### PR TITLE
OSDOCS-10271: VMware CPMS TP to GA

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.16+

Issue:
[OSDOCS-10271](https://issues.redhat.com//browse/OSDOCS-10271)

Link to docs preview:
- [Supported cloud providers](https://75106--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-getting-started.html#cpmso-platform-matrix_cpmso-getting-started)
- [Sample VMware vSphere failure domain configuration](https://75106--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-configuration.html#cpmso-yaml-failure-domain-vsphere_cpmso-configuration)

QE review:
- [x] QE has approved this change.

Additional information:
There is install impact too, this is just for the machine mgmt book
RN update: #75109

### CODE REVIEW

**Comment by jeana-redhat** [comment]
@rvanderp3 this is the part of the VMW CPMS stuff that lives in the _Machine management_ book

Rel notes for same are here: #75109

PTAL :bow:

File: modules/cpmso-yaml-failure-domain-vsphere.adoc [hunk 1]
```diff
@@ -4,21 +4,21 @@
 
 :_mod-docs-content-type: REFERENCE
 [id="cpmso-yaml-failure-domain-vsphere_{context}"]
-= Sample VMware vSphere failure domain configuration
+= Sample {vmw-full} failure domain configuration
 
-On VMware vSphere infrastructure, the cluster-wide infrastructure Custom Resource Definition (CRD), `infrastructures.config.openshift.io`, defines failure domains for your cluster. The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains. A failure domain is an infrastructure resource that comprises a control plane machine set, a vCenter datacenter, vCenter datastore, and a network.
+On {vmw-full} infrastructure, the cluster-wide infrastructure Custom Resource Definition (CRD), `infrastructures.config.openshift.io`, defines failure domains for your cluster.
+The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains.
```
Comment (rvanderp3) [inline modules/cpmso-yaml-failure-domain-vsphere.adoc:10]:
```suggestion
The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains which the control plane machineset will use to ensure control plane nodes are deployed to the appropriate failure domain.
```

File: modules/cpmso-yaml-failure-domain-vsphere.adoc [hunk 2]
```diff
@@ -4,21 +4,21 @@
...
+A failure domain is an infrastructure resource made up of a control plane machine set, a vCenter datacenter, vCenter datastore, and a network.
 
-By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on hardware that is separate from the primary VMware vSphere infrastructure. A control plane machine set also balances control plane machines across defined failure domains to provide fault tolerance capabilities to your infrastructure.
+By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on hardware that is separate from the primary {vmw-full} infrastructure.
```
Comment (rvanderp3) [inline modules/cpmso-yaml-failure-domain-vsphere.adoc:13]:
```suggestion
By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on  separate clusters or datacenters.  
```

**Review by rvanderp3 — COMMENTED** [review]
overall lgtm, i found a couple of items that may need to be reworded.

**Comment by jeana-redhat** [comment]
> overall lgtm, i found a couple of items that may need to be reworded.

Thanks! updated.

File: modules/cpmso-yaml-failure-domain-vsphere.adoc [hunk 3]
```diff
@@ -43,5 +43,6 @@ spec:
 
 [IMPORTANT]
 ====
-Each `failureDomains.platform.vsphere.name` field value in the `ControlPlaneMachineSet` CR must match the corresponding value defined in the `failureDomains.name` field of the cluster-wide infrastructure CRD. Currently, the `vsphere.name` field is the only supported failure domain field that you can specify in the `ControlPlaneMachineSet` CR.
-====
+Each `failureDomains.platform.vsphere.name` field value in the `ControlPlaneMachineSet` CR must match the corresponding value defined in the `failureDomains.name` field of the cluster-wide infrastructure CRD.
+Currently, the `vsphere.name` field is the only supported failure domain field that you can specify in the `ControlPlaneMachineSet` CR.
+====
```
Comment (miyadav) [inline modules/cpmso-yaml-failure-domain-vsphere.adoc:48]: LGTM , if it helps we can add below  to get failure domain name from infrastructure CRD.
`oc get infrastructure cluster -o=jsonpath={.spec.platformSpec.vsphere.failureDomains[0].name}`

Comment (jeana-redhat) [reply modules/cpmso-yaml-failure-domain-vsphere.adoc:48]: That's a great thing to add, thanks!

**Comment by tmalove** [comment]
@jeana-redhat  
/lgtm

**Pull request merged**